### ISSUE TITLE
Directory not readable - File system cannot find the path

### ISSUE THREAD

**Issue by imsumoregn** [wez/wezterm#3916]
### What Operating System(s) are you seeing this problem on?

Windows

### WezTerm version

wezterm 20230629-080200-f376468f

### Did you try the latest nightly build to see if the issue is better (or worse!) than your current version?

Yes, and I updated the version box above to show the version of the nightly that I tried

### Describe the bug

This is another issue regarding the not-working pane-splitting feature, but with a little bit more investigation, and a (potential) workaround.

So, when I tried splitting it, this came out.
```
Directory "..." is not readable and will not be used for the command we are spawning: The filename, directory name, or volume label syntax is incorrect. (os error 123)
```

I just cannot give up at this point; and for the love of God, here are all possible ways I could think of to split the pane.
```lua
wezterm.on('sv', function(window, pane)
  local cwd = pane:get_current_working_dir()
  wezterm.log_info(cwd)
  window:perform_action(wezterm.action.SplitVertical({ cwd = cwd, domain = 'CurrentPaneDomain' }), pane)
end)

config.keys = {
  {
    key = '-',
    mods = 'LEADER',
    action = wezterm.action.SplitVertical { domain = 'CurrentPaneDomain' },
    -- action = wezterm.action_callback(function(window, pane)
    --   local cwd = pane:get_current_working_dir()
    --   window:perform_action(wezterm.action.SplitVertical({ cwd = cwd, domain = 'CurrentPaneDomain' }), pane)
    -- end),
    -- action = wezterm.action.EmitEvent 'sv',
  }
}
```

2.5 cups of coffee made me mutate the `cwd`.
```lua
-- Using built-in actions give you '/home/harry/work'
action = wezterm.action.SplitVertical { domain = 'CurrentPaneDomain' }

-- Using `window:perform_action` or emitting an event gives you 'file://local/home/harry/work'
action = wezterm.action_callback(function(window, pane)
  local cwd = pane:get_current_working_dir()
  window:perform_action(wezterm.action.SplitVertical({ cwd = cwd, domain = 'CurrentPaneDomain' }), pane)
end),
action = wezterm.action.EmitEvent 'sv'

-- BUT, if you mutate `cwd` like this,
action = wezterm.action_callback(function(window, pane)
  local cwd = pane:get_current_working_dir()

  cwd = 'C:\\Users\\harry'

  window:perform_action(wezterm.action.SplitVertical({ cwd = cwd, domain = 'CurrentPaneDomain' }), pane)
end),
-- all newly-split panes will be at 'C:\\Users\\harry'!
```

It would be lovely if I could learn how `wezterm` accepts the Windows-style path for normal drives but not for `wsl`, because I tried.
```lua
cwd = '\\wsl$:\\Ubuntu-20.04\\home\\harry\\work'
cwd = '\\wsl.localhost\\Ubuntu-20.04\\home\\harry\\work'
-- And a few other variants constructed by randomly removing/adding characters hopelessly.
```

### Configuration

No config.

**Comment by wez** [comment]
It sounds like maybe you're using WSL, but it's not clear whether you are using WslDomains or something else.
The configuration and steps you've shared are incomplete, so there's not a lot to go on.
Please add the smallest configuration that reproduces the issue, along with precise steps to follow.

**Comment by imsumoregn** [comment]
My bad, Wez, at first I was using WSL with no configuration as I said, but then after adding
```lua
config.default_domain = 'WSL:Ubuntu-22.04'
```
everything works as expected. I could only talk on behalf of myself, but the above line could be a great help if existed in the `Launching Programs` page, since the only thing I found that is somewhat related to setting up `wezterm` with wsl is
```lua
-- Spawn a fish shell in login mode
config.default_prog = { '/usr/local/bin/fish', '-l' }
```
which after modifying to
```lua
-- Spawn a fish shell in login mode
config.default_prog = { 'wsl' }
```
`wezterm` is launched in wsl, but then trying to split, the `cwd` is always `%USERPROFILE%`.

Closing on this note, great work with this fast boy, Wez; I enjoyed it. Have a nice day.
